src/main/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerTimeEntry.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.homeoffice.digital.sas.timecard.producers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.stereotype.Component;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+@Component
+@Slf4j
+public class KafkaProducerTimeEntry {
+
+  private final KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+
+  public KafkaProducerTimeEntry(KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate) {
+    this.kafkaTimeEntryTemplate = kafkaTimeEntryTemplate;
+  }
+
+  public void sendMessage(TimeEntry timeEntry, KafkaAction action) {
+    KafkaEventMessage kafkaEventMessage = new KafkaEventMessage(timeEntry, action);
+    ListenableFuture<SendResult<String, KafkaEventMessage>> future = kafkaTimeEntryTemplate.send(
+        "callisto-timecard",
+        timeEntry.getOwnerId().toString(),
+        kafkaEventMessage
+    );
+
+    listenableFutureReporting(timeEntry, kafkaEventMessage, future);
+  }
+
+  private static void listenableFutureReporting(TimeEntry timeEntry,
+                                                KafkaEventMessage kafkaEventMessage,
+                                                ListenableFuture<SendResult<String,
+                                                    KafkaEventMessage>> future) {
+    future.addCallback(new ListenableFutureCallback<>() {
+
+      @Override
+      public void onFailure(Throwable ex) {
+        log.error(String.format("Sent message has failed=[ %s ]",
+            kafkaEventMessage), ex);
+      }
+
+      @Override
+      public void onSuccess(SendResult<String, KafkaEventMessage> result) {
+        log.info(String.format("Sent message=[ %s ]", timeEntry));
+      }
```

Comment:
It might be worth while testing that we log success and failure messages. Probs not a major requirement but might be good if not time consuming. I think you can achieve this using the slf4j `LoggerFactory`.

Reply:
As discussed - we found it difficult to test these logs as they are not directly called, so we have decided not to test them